packages/manager/apps/pci-ai-notebooks/src/pages/Auth.layout.tsx

```diff
@@ -0,0 +1,36 @@
+import { Outlet, redirect } from 'react-router-dom';
+import Breadcrumb from '@/components/breadcrumb/Breadcrumb.component';
+import queryClient from '@/query.client';
+import { getAuthorization } from '@/data/api/ai/authorization.api';
+
+interface AuthLayoutProps {
+  params: {
+    projectId: string;
+  };
+  request: Request;
+}
+// check if the user has authentication
+export const Loader = async ({ params }: AuthLayoutProps) => {
+  const { projectId } = params;
+  try {
+    const authResult = await queryClient.fetchQuery({
+      queryKey: [projectId, 'ai/authorization'],
+      queryFn: () => getAuthorization({ projectId }),
+    });
+    if (!authResult.authorized) {
+      return redirect(`/pci/projects/${projectId}/ai/notebooks/auth`);
+    }
+  } catch (_error) {
+    return redirect(`/pci/projects/${projectId}/ai/notebooks/auth`);
```

Comment:
Are you sure that redirecting to the auth page even when the call fails is the intended behavior here ? I'd suggest throwing the error so it can be intercepted by the errorBoundary and displays the error page. To do so, you could simply remove the try catch block, like you did in the Root.page loader function for the notebooks

Reply:
ok, done